### ISSUE TITLE
Duplicate operationIds

### ISSUE THREAD

**Issue by knutwannheden** [gigaSproule/swagger-gradle-plugin#182]
It appears that this plugin has the same problem the Maven plugin had a while back: kongchen/swagger-maven-plugin#377.

According to the specification the `operationId` must be unique within the whole API. Currently the `operationId` is derived from the method name only and will thus result in duplicates if the API has multiple annotated Java methods with the same name: https://github.com/gigaSproule/swagger-gradle-plugin/blob/3e274f32642ad4573ecfb7856b738f9cd7ec5ee2/plugin/src/main/groovy/com/benjaminsproule/swagger/gradleplugin/reader/JaxrsReader.groovy#L258

I have seen other implementations where a number suffix will be used (e.g. `foo` and `foo_1`). In the Maven plugin there is a configuration parameter with which the format can be defined. E.g. `<operationIdFormat>{{className}}_{{methodName}}_{{httpMethod}}</operationIdFormat>`.

**Comment by dcp65** [comment]
Using the `nickname` attribute with the Swagger `ApiOperation` annotation ensures that each operation has a unique name.

**Comment by knutwannheden** [comment]
Yes, I have seen that. But I think it would make sense if the generated `operationId` would also be unique, as with the Maven plugin. What do you think?

**Comment by gigaSproule** [comment]
@knutwannheden I agree this does make sense. Just for future reference, this default is applied in `JaxrsReader#parseMethod` and `SpringMvcReader#parseMethod`, so should be an easy update.

**Comment by anichari** [comment]
Any updates on this issue so far? We too are facing similar duplidate operationIds issues. It would be really useful to be able to configure the format like we do in the maven plugin
